docs/trace/README.md

```diff
@@ -1,19 +1,122 @@
+singleton via dependency injection (e.g.
+[Instrumentation.cs](../../examples/AspNetCore/Instrumentation.cs)).
+
+:heavy_check_mark: You should use dot-separated
+[UpperCamelCase](https://en.wikipedia.org/wiki/Camel_case) as the
+[`ActivitySource.Name`](https://learn.microsoft.com/dotnet/api/system.diagnostics.activitysource.name).
+In many cases, using the fully qualified class name might be a good option.
+
+```csharp
+static readonly ActivitySource MyActivitySource = new("MyCompany.MyProduct.MyLibrary");
+```
+
+:heavy_check_mark: You should check
+[`Activity.IsAllDataRequested`](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity.isalldatarequested)
+before [setting
+Tags](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity.settag)
+for better performance.
+
+```csharp
+using (var activity = MyActivitySource.StartActivity("SayHello"))
+{
+    if (activity != null && activity.IsAllDataRequested == true)
+    {
+        activity.SetTag("http.url", "http://www.mywebsite.com");
+    }
+}
+```
+
+:heavy_check_mark: You should use
+[Activity.SetTag](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity.settag)
+instead of
+[Activity.AddTag](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity.addtag)
+and
+[Activity.SetCustomProperty](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity.setcustomproperty)
+because the latter do not perform deduplication.
+
+## TracerProvider Management
+
+:stop_sign: You should avoid creating `TracerProvider` instances too frequently,
+`TracerProvider` is fairly expensive and meant to be reused throughout the
+application. For most applications, one `TracerProvider` instance per process
+would be sufficient.
+
+:heavy_check_mark: You should properly manage the lifecycle of `TracerProvider`
+instances if they are created by you.
+
+Here is the rule of thumb when managing the lifecycle of `TracerProvider`:
+
+* If you are building an application with [dependency injection
+  (DI)](https://learn.microsoft.com/dotnet/core/extensions/dependency-injection)
+  (e.g. [ASP.NET Core](https://learn.microsoft.com/aspnet/core) and [.NET
+  Worker](https://learn.microsoft.com/dotnet/core/extensions/workers)), in most
+  cases you should create the `TracerProvider` instance and let DI manage its
+  lifecycle. Refer to the [Getting Started with OpenTelemetry .NET Traces in 5
+  Minutes - ASP.NET Core Application](./getting-started-aspnetcore/README.md)
+  tutorial to learn more.
+* If you are building an application without DI, create a `TracerProvider`
+  instance and manage the lifecycle explicitly. Refer to the [Getting Started
+  with OpenTelemetry .NET Traces in 5 Minutes - Console
+  Application](./getting-started-console/README.md) tutorial to learn more.
+* If you forget to dispose the `TracerProvider` instance before the application
+  ends, activities might get dropped due to the lack of proper flush.
```

Comment:
```suggestion
  ends, finished activities might get dropped due to the lack of proper flush.
```

^ This is good to follow up separately, as we dont have the content yet mentioning started-but-not-yet activities, and how to avoid that etc..

Reply:
Will handle it in a separate PR.